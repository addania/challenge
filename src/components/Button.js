import React from "react";
import "./Button.css";

export function Button({ onClick }) {
  return (
    <button
      type="submit"
      style={{
        backgroundColor: "#8DA1B9",
        color: "white",
        borderRadius: "2px",
        width: "80px",
        padding: "5px 0px",
        position: "left",
        marginTop: "10px"
      }}
    >
      <span onClick={onClick}>Apply</span>
    </button>
  );
}
