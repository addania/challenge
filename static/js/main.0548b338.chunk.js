(this.webpackJsonpchallenge=this.webpackJsonpchallenge||[]).push([[0],{155:function(e,t,n){e.exports=n(282)},160:function(e,t,n){},162:function(e,t,n){},281:function(e,t,n){},282:function(e,t,n){"use strict";n.r(t);var a=n(0),l=n.n(a),r=n(31),o=n.n(r),s=(n(160),n(16)),c=n(68),i=n(69),u=n.n(i),m=n(21),p=(n(162),n(48),n(18)),g=n.n(p),d=n(289),f=n(290),h=n(291);function v(){return l.a.createElement("div",null,l.a.createElement(d.a,null,l.a.createElement(f.a,{sm:12},l.a.createElement("h2",{style:{margin:"20px 0px",textAlign:"left",color:"#7F6A93",borderBottom:"0.5px solid #E8E8EA",paddingBottom:"20px"}},"Advertising Data ETL-V Challenge"))))}function E(e){return l.a.createElement("li",null,e.message,l.a.createElement("span",{style:e.styling},e.columns,"s"))}function b(e){for(var t={fontStyle:"italic"},n=[],a=0;a<e.dimensionsColumns.length;a++)n.push(l.a.createElement(E,{message:"Select zero to N ",columns:e.dimensionsColumns[a],styling:t,key:a}));return l.a.createElement("div",null,l.a.createElement(d.a,null,l.a.createElement(f.a,{sm:12},l.a.createElement("div",{style:{margin:"20px 0px",textAlign:"left",color:"#828282",fontSize:"15px",borderBottom:"0.5px solid #E8E8EA",paddingBottom:"20px"}},l.a.createElement("ul",{style:{paddingLeft:"20px"}},n),l.a.createElement("p",{style:{fontSize:"12px"}},'[where zero means "All"]'),l.a.createElement("p",null,'Hitting "Apply" filters the chart to show a timeseries for both'," ",l.a.createElement("span",{style:t},e.metricsColumns[0])," and"," ",l.a.createElement("span",{style:t},e.metricsColumns[1])," for given"," ",l.a.createElement("span",{style:t},e.dimensionsColumns[0],"s")," and"," ",l.a.createElement("span",{style:t},e.dimensionsColumns[1],"s")," - logical AND")))))}var y=n(32),x=n.n(y),O=n(130),C=n.n(O);function j(e){var t,n=e.coreData,a=e.filters,r=e.applyFilters,o=e.onClick,c=e.styling,i=function(e,t,n){var a,l,r;if(n){var o=function(e,t){Object.values(e);var n=e.Datasource,a=e.Campaign,l=[];0===Object.entries(e).length&&e.constructor===Object||void 0!==n&&0!=n.length?0===Object.entries(e).length&&e.constructor===Object||void 0!==a&&0!=a.length?0===Object.entries(e).length&&e.constructor===Object||(l=g.a.filter(t,(function(e){return n.includes(e.Datasource)&&a.includes(e.Campaign)}))):l=g.a.filter(t,(function(e){return n.includes(e.Datasource)})):l=g.a.filter(t,(function(e){return a.includes(e.Campaign)}));return l}(t,e);console.log("arrayAfterFiltering",o);var s=k(o);console.log("aggregatedData",s),a=s[0],console.log("aggregatedImpressions",a),l=s[1],console.log("aggregatedClicks",l),r=s[2]}else{var c=k(e);console.log("aggregatedData",c),a=c[0],console.log("aggregatedImpressions",a),l=c[1],console.log("aggregatedClicks",l),r=c[2],console.log("aggregatedDates",r)}return i=a,u=l,m=r,{chart:{type:"spline"},xAxis:{categories:m},yAxis:[{labels:{format:"{value}",style:{color:x.a.getOptions().colors[0]}},title:{text:"Clicks",style:{color:x.a.getOptions().colors[0]}},opposite:!1},{gridLineWidth:0,title:{text:"Impressions",style:{color:x.a.getOptions().colors[1]}},labels:{format:"{value}",style:{color:x.a.getOptions().colors[1]}},opposite:!0}],title:{text:"Datasources",align:"left"},series:[{name:"Clicks",data:u},{name:"Impressions",yAxis:1,data:i}]};var i,u,m}(n,a,r);return l.a.createElement("div",null,l.a.createElement(d.a,null,l.a.createElement(f.a,{sm:10},l.a.createElement("h2",{style:c},"Chart")),l.a.createElement(f.a,{sm:2},l.a.createElement("button",{type:"submit",style:(t={backgroundColor:"#8DA1B9",color:"white",borderRadius:"2px",width:"80px",padding:"5px 0px",position:"left",marginTop:"10px"},Object(s.a)(t,"position","absolute"),Object(s.a)(t,"right",0),Object(s.a)(t,"bottom",0),t)},l.a.createElement("span",{onClick:o},"Reset")))),l.a.createElement(C.a,{highcharts:x.a,options:i}))}function k(e){console.log("array",e);var t=g.a.groupBy(e,"Date");console.log("groupByDate",t);for(var n=g.a.keys(t),a=[],l=[],r=0;r<n.length;r++){for(var o=0,s=0,c=0;c<t[n[r]].length;c++)o+=t[n[r]][c].Impressions,s+=t[n[r]][c].Clicks;console.log("totalImpressions",o),console.log("totalClicks",s),a.push(o),l.push(s)}return[a,l,n]}var D=n(292);function A(e){var t=e.styling,n=e.filterColumns,r=e.onChange,o=e.dataSet,c=Object(a.useState)(),i=Object(m.a)(c,2),u=(i[0],i[1],[]),p=document.createElement("link");p.rel="stylesheet",p.href="https://cdn.jsdelivr.net/npm/semantic-ui/dist/semantic.min.css",document.head.appendChild(p);for(var g=0;g<n.length;g++){var d=w(n[g],o);u.push(l.a.createElement("div",{key:g},l.a.createElement("p",{style:{textAlign:"left",color:"#828282",marginTop:"20px"}},n[g]),l.a.createElement(D.a,Object(s.a)({key:g,style:p,placeholder:n[g],fluid:!0,multiple:!0,selection:!0,options:d,onChange:r},"style",{marginTop:"5px"}))))}return l.a.createElement("div",null,l.a.createElement("h2",{style:t},"Filters"),u)}function w(e,t){for(var n=[],a=0;a<t.length;a++)n.includes(t[a][e])||n.push(t[a][e]);for(var l=[],r=0;r<n.length;r++){var o={key:n[r],text:n[r],value:n[r]};l.push(o)}return l}n(281);function B(e){var t=e.onClick;return l.a.createElement("button",{type:"submit",style:{backgroundColor:"#8DA1B9",color:"white",borderRadius:"2px",width:"80px",padding:"5px 0px",position:"left",marginTop:"10px"}},l.a.createElement("span",{onClick:t},"Apply"))}var S=function(){var e=Object(a.useState)([]),t=Object(m.a)(e,2),n=t[0],r=t[1],o=Object(a.useState)([]),i=Object(m.a)(o,2),p=i[0],g=i[1],E=Object(a.useState)([]),y=Object(m.a)(E,2),x=y[0],O=y[1],C=Object(a.useState)([]),k=Object(m.a)(C,2),D=(k[0],k[1]),w=Object(a.useState)([]),S=Object(m.a)(w,2),J=(S[0],S[1],Object(a.useState)({})),H=Object(m.a)(J,2),K=H[0],M=H[1],V=Object(a.useState)(!1),Y=Object(m.a)(V,2),$=Y[0],q=Y[1];Object(a.useEffect)((function(){!function(){var e,t,n,a,l,o,s,c,i,m;u.a.async((function(p){for(;;)switch(p.prev=p.next){case 0:return p.next=2,u.a.awrap(fetch("http://adverity-challenge.s3-website-eu-west-1.amazonaws.com/DAMKBAoDBwoDBAkOBAYFCw.csv"));case 2:return e=p.sent,p.next=5,u.a.awrap(e.text());case 5:t=p.sent,n=I(t),a=F(n),l=z(a),o=T(l),s=N(o[0]),c=L(o[0],s),i=R(o[0],s),m=W(o[0],s),r(o),g(c),O(i),D(m);case 18:case"end":return p.stop()}}))}()}),[]);var G={margin:"30px 0px",textAlign:"left",color:"#8DA1B9"};return l.a.createElement("div",{className:"App"},l.a.createElement(h.a,null,l.a.createElement(v,null),l.a.createElement(b,{dimensionsColumns:x,metricsColumns:p}),l.a.createElement(d.a,null,l.a.createElement(f.a,{sm:4,style:{backgroundColor:"#EFF2F1"}},l.a.createElement(A,{styling:G,filterColumns:x,onChange:function(e,t){if(q(!1),void 0===t.value||0==t.value){var n=Object(c.a)({},K);delete n[t.placeholder],M(n)}else{var a=Object(c.a)({},K),l=t.placeholder;a=Object.assign(a,Object(s.a)({},l,t.value)),M(a)}},dataSet:n}),l.a.createElement(B,{onClick:function(){q(!0)}})),l.a.createElement(f.a,{sm:8},l.a.createElement(j,{coreData:n,filters:K,applyFilters:$,onClick:function(){q(!1)},styling:G})))))};function I(e){for(var t=e.split("\n"),n=[],a=t[0].split(","),l=1;l<t.length;l++){for(var r={},o=t[l].split(","),s=0;s<a.length;s++){var c=o[s];r[a[s]]=3==s||4==s?parseInt(c):c}n.push(r)}return n}function F(e){for(var t=0;t<e.length;t++)e[t].Impressions||(e[t].Impressions=0),e[t].Clicks||(e[t].Clicks=0);return e}function z(e){for(var t=[],n=0;n<e.length;n++){var a=Object(c.a)({},e[n]),l=e[n].Date,r=l.slice(6,10),o=l.slice(3,5),s=l.slice(0,2),i=new Date(r+"-"+o+"-"+s);a.Date=i,t.push(a)}return t}function T(e){return g.a.sortBy(e,["Date","Datasource"])}function N(e){return Object.keys(e)}function L(e,t){for(var n=[],a=0;a<t.length;a++){var l=t[a];g.a.isNumber(e[l])&&n.push(l)}return n}function R(e,t){for(var n=[],a=0;a<t.length;a++){var l=t[a];g.a.isString(e[l])&&n.push(l)}return n}function W(e,t){for(var n=[],a=0;a<t.length;a++){var l=t[a];g.a.isDate(e[l])&&n.push(l)}return n}Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));o.a.render(l.a.createElement(S,null),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()}))}},[[155,1,2]]]);
//# sourceMappingURL=main.0548b338.chunk.js.map