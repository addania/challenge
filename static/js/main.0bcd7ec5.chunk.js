(this.webpackJsonpchallenge=this.webpackJsonpchallenge||[]).push([[0],{154:function(e,t,a){e.exports=a(279)},159:function(e,t,a){},160:function(e,t,a){},279:function(e,t,a){"use strict";a.r(t);var n=a(0),l=a.n(n),s=a(29),r=a.n(s),o=(a(159),a(127)),i=a(32),c=(a(160),a(70),a(286)),m=a(287),u=a(288);function p(){return l.a.createElement("div",null,l.a.createElement(c.a,null,l.a.createElement(m.a,{sm:12},l.a.createElement("h2",{style:{margin:"30px 0px",textAlign:"left",color:"#7F6A93",borderBottom:"0.5px solid #E8E8EA",paddingBottom:"20px"}},"Advertising Data ETL-V Challenge"))))}function d(e){return l.a.createElement("li",null,e.message,l.a.createElement("span",{style:e.styling},e.columns,"s"))}function g(e){for(var t={fontStyle:"italic"},a=[],n=0;n<e.dimensionsColumns.length;n++)a.push(l.a.createElement(d,{message:"Select zero to N ",columns:e.dimensionsColumns[n],styling:t,key:n}));return l.a.createElement("div",null,l.a.createElement(c.a,null,l.a.createElement(m.a,{sm:12},l.a.createElement("div",{style:{margin:"30px 0px",textAlign:"left",color:"#828282",fontSize:"18px",borderBottom:"0.5px solid #E8E8EA",paddingBottom:"20px"}},l.a.createElement("ul",{style:{paddingLeft:"20px"}},a),l.a.createElement("p",{style:{fontSize:"12px"}},'[where zero means "All"]'),l.a.createElement("p",null,'Hitting "Apply" filters the chart to show a timeseries for both ',l.a.createElement("span",{style:t},e.metricsColumns[0])," and ",l.a.createElement("span",{style:t},e.metricsColumns[1])," for given ",l.a.createElement("span",{style:t},e.dimensionsColumns[0],"s")," and ",l.a.createElement("span",{style:t},e.dimensionsColumns[1],"s")," - logical AND")))))}var h=a(30),f=a.n(h),E=a(129),v=a.n(E);function y(e){var t={chart:{type:"spline"},xAxis:{categories:e.data},yAxis:[{labels:{format:"{value}",style:{color:f.a.getOptions().colors[0]}},title:{text:"Clicks",style:{color:f.a.getOptions().colors[0]}},opposite:!1},{gridLineWidth:0,title:{text:"Impressions",style:{color:f.a.getOptions().colors[1]}},labels:{format:"{value}",style:{color:f.a.getOptions().colors[1]}},opposite:!0}],title:{text:"Datasources",align:"left"},series:[{name:"Clicks",data:[12,4,3,4,4]},{name:"Impressions",yAxis:1,data:[10,20,30,40,30]}]};return l.a.createElement("div",null,l.a.createElement("h2",{style:e.styling},"Chart"),l.a.createElement(v.a,{highcharts:f.a,options:t}))}var C=a(289);function D(e){var t=e.styling,a=e.filterColumns,s=e.onChange,r=e.dataSet,o=Object(n.useState)(),c=Object(i.a)(o,2),m=(c[0],c[1],[]),u=document.createElement("link");u.rel="stylesheet",u.href="https://cdn.jsdelivr.net/npm/semantic-ui/dist/semantic.min.css",document.head.appendChild(u);for(var p=0;p<a.length;p++){var d=x(a[p],r);console.log("genOptions",d),m.push(l.a.createElement(C.a,{style:u,placeholder:a[p],fluid:!0,multiple:!0,selection:!0,options:d,onChange:s}))}return l.a.createElement("div",null,l.a.createElement("h2",{style:t},"Filters"),m)}function x(e,t){for(var a=[],n=0;n<t.length;n++)a.includes(t[n][e])||a.push(t[n][e]);for(var l=[],s=0;s<a.length;s++){var r={key:a[s],text:a[s],value:a[s]};l.push(r)}return l}var A=a(46),b=a.n(A);var O=function(){var e,t=Object(n.useState)([{Date:"01.01.2019",Datasource:"Facebook Ads",Campaign:"Like Ads",Clicks:274,Impressions:1979},{Date:"01.01.2019",Datasource:"Facebook Ads",Campaign:"Offer Campaigns - Conversions",Clicks:10245,Impressions:764627},{Date:"01.01.2019",Datasource:"Google Adwords",Campaign:"B2B - Leads",Clicks:7,Impressions:444},{Date:"01.01.2019",Datasource:"Google Adwords",Campaign:"GDN Prospecting - App - Prio 1 Offer",Clicks:16,Impressions:12535},{Date:"20.01.2019",Datasource:"Google Adwords",Campaign:"GDN Prospecting - App - Prio 1 Offer",Clicks:25,Impressions:10535}]),a=Object(i.a)(t,2),s=a[0],r=(a[1],Object(n.useState)([])),d=Object(i.a)(r,2),h=(d[0],d[1],Object(n.useState)([])),f=Object(i.a)(h,2),E=f[0],v=f[1],C=Object(n.useState)([]),x=Object(i.a)(C,2),A=x[0],O=x[1],k={margin:"30px 0px",textAlign:"left",color:"#8DA1B9"},w=(e=s,Object.keys(e[0])),S=function(e){return b.a.sortBy(e,["Date","Datasource"])}(function(e){for(var t=[],a=0;a<e.length;a++){var n=Object(o.a)({},e[a]),l=e[a].Date,s=l.slice(6,10),r=l.slice(3,5),i=l.slice(0,2),c=new Date(s+"-"+r+"-"+i);n.Date=c,t.push(n)}return t}(s)),j=function(e,t){for(var a=[],n=0;n<t.length;n++){var l=t[n];b.a.isNumber(e[0][l])&&a.push(l)}return a}(S,w),B=function(e,t){for(var a=[],n=0;n<t.length;n++){var l=t[n];b.a.isString(e[0][l])&&a.push(l)}return a}(S,w);return function(e,t){for(var a=[],n=0;n<t.length;n++){var l=t[n];b.a.isDate(e[0][l])&&a.push(l)}}(S,w),S[0].Date,S[1].Date,S[2].Date,S[3].Date,S[4].Date,console.log("selected values",E),l.a.createElement("div",{className:"App"},l.a.createElement(u.a,null,l.a.createElement(p,null),l.a.createElement(g,{dimensionsColumns:B,metricsColumns:j}),l.a.createElement(c.a,null,l.a.createElement(m.a,{sm:4},l.a.createElement(D,{styling:k,filterColumns:B,onChange:function(e,t){"Datasource"==t.placeholder?v(t.value):"Campaign"==t.placeholder&&O(t.value)},dataSet:S}),l.a.createElement("p",null,"Selected values Datasource: ",E),l.a.createElement("p",null,"Selected values Campaigns: ",A)),l.a.createElement(m.a,{sm:8},l.a.createElement(y,{styling:k,coreData:S})))))};Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));r.a.render(l.a.createElement(O,null),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()}))}},[[154,1,2]]]);
//# sourceMappingURL=main.0bcd7ec5.chunk.js.map