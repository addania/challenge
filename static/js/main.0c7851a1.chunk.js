(this.webpackJsonpchallenge=this.webpackJsonpchallenge||[]).push([[0],{154:function(e,t,a){e.exports=a(279)},159:function(e,t,a){},160:function(e,t,a){},279:function(e,t,a){"use strict";a.r(t);var n=a(0),l=a.n(n),o=a(29),s=a.n(o),r=(a(159),a(127)),i=a(42),c=(a(160),a(70),a(286)),u=a(287),m=a(288);function p(){return l.a.createElement("div",null,l.a.createElement(c.a,null,l.a.createElement(u.a,{sm:12},l.a.createElement("h2",{style:{margin:"30px 0px",textAlign:"left",color:"#7F6A93",borderBottom:"0.5px solid #E8E8EA",paddingBottom:"20px"}},"Advertising Data ETL-V Challenge"))))}function g(e){return l.a.createElement("li",null,e.message,l.a.createElement("span",{style:e.styling},e.columns,"s"))}function d(e){for(var t={fontStyle:"italic"},a=[],n=0;n<e.dimensionsColumns.length;n++)a.push(l.a.createElement(g,{message:"Select zero to N ",columns:e.dimensionsColumns[n],styling:t,key:n}));return l.a.createElement("div",null,l.a.createElement(c.a,null,l.a.createElement(u.a,{sm:12},l.a.createElement("div",{style:{margin:"30px 0px",textAlign:"left",color:"#828282",fontSize:"18px",borderBottom:"0.5px solid #E8E8EA",paddingBottom:"20px"}},l.a.createElement("ul",{style:{paddingLeft:"20px"}},a),l.a.createElement("p",{style:{fontSize:"12px"}},'[where zero means "All"]'),l.a.createElement("p",null,'Hitting "Apply" filters the chart to show a timeseries for both ',l.a.createElement("span",{style:t},e.metricsColumns[0])," and ",l.a.createElement("span",{style:t},e.metricsColumns[1])," for given ",l.a.createElement("span",{style:t},e.dimensionsColumns[0],"s")," and ",l.a.createElement("span",{style:t},e.dimensionsColumns[1],"s")," - logical AND")))))}var v=a(30),h=a.n(v),y=a(129),f=a.n(y);function E(e){var t={chart:{type:"spline"},xAxis:{categories:e.data},yAxis:[{labels:{format:"{value}",style:{color:h.a.getOptions().colors[0]}},title:{text:"Clicks",style:{color:h.a.getOptions().colors[0]}},opposite:!1},{gridLineWidth:0,title:{text:"Impressions",style:{color:h.a.getOptions().colors[1]}},labels:{format:"{value}",style:{color:h.a.getOptions().colors[1]}},opposite:!0}],title:{text:"Datasources",align:"left"},series:[{name:"Clicks",data:[12,4,3,4,4]},{name:"Impressions",yAxis:1,data:[10,20,30,40,30]}]};return l.a.createElement("div",null,l.a.createElement("h2",{style:e.styling},"Chart"),l.a.createElement(f.a,{highcharts:h.a,options:t}))}var x=a(289);function k(e){var t=e.styling,a=e.filterColumns,o=e.onChange,s=Object(n.useState)(),r=Object(i.a)(s,2),c=r[0];r[1];console.log("filterColumns",a);for(var u=[],m=function(e){u.push(l.a.createElement("button",{key:e,name:a[e],onClick:function(){return onclick(a[e])}},a[e])),console.log("columns",u)},p=0;p<a.length;p++)m(p);console.log("columns",u);var g=document.createElement("link");return g.rel="stylesheet",g.href="https://cdn.jsdelivr.net/npm/semantic-ui/dist/semantic.min.css",document.head.appendChild(g),l.a.createElement("div",null,l.a.createElement("h2",{style:t},"Selected values are: ",c),u,l.a.createElement(x.a,{style:g,placeholder:"Skills",fluid:!0,multiple:!0,selection:!0,options:[{key:"angular",text:"Angular",value:"angular",name:"Angular"},{key:"css",text:"CSS",value:"css"},{key:"design",text:"Graphic Design",value:"design"},{key:"ember",text:"Ember",value:"ember"},{key:"html",text:"HTML",value:"html"},{key:"ia",text:"Information Architecture",value:"ia"},{key:"javascript",text:"Javascript",value:"javascript"},{key:"mech",text:"Mechanical Engineering",value:"mech"},{key:"meteor",text:"Meteor",value:"meteor"},{key:"node",text:"NodeJS",value:"node"},{key:"plumbing",text:"Plumbing",value:"plumbing"},{key:"python",text:"Python",value:"python"},{key:"rails",text:"Rails",value:"rails"},{key:"react",text:"React",value:"react"},{key:"repair",text:"Kitchen Repair",value:"repair"},{key:"ruby",text:"Ruby",value:"ruby"},{key:"ui",text:"UI Design",value:"ui"},{key:"ux",text:"User Experience",value:"ux"}],onChange:o}))}var C=a(46),b=a.n(C);var D=function(){var e=Object(n.useState)([{Date:"01.01.2019",Datasource:"Facebook Ads",Campaign:"Like Ads",Clicks:274,Impressions:1979},{Date:"01.01.2019",Datasource:"Facebook Ads",Campaign:"Offer Campaigns - Conversions",Clicks:10245,Impressions:764627},{Date:"01.01.2019",Datasource:"Google Adwords",Campaign:"B2B - Leads",Clicks:7,Impressions:444},{Date:"01.01.2019",Datasource:"Google Adwords",Campaign:"GDN Prospecting - App - Prio 1 Offer",Clicks:16,Impressions:12535},{Date:"20.01.2019",Datasource:"Google Adwords",Campaign:"GDN Prospecting - App - Prio 1 Offer",Clicks:25,Impressions:10535}]),t=Object(i.a)(e,2),a=t[0],o=(t[1],Object(n.useState)([])),s=Object(i.a)(o,2),g=(s[0],s[1],Object(n.useState)([])),v=Object(i.a)(g,2),h=v[0],y=v[1],f={margin:"30px 0px",textAlign:"left",color:"#8DA1B9"};console.log(JSON.stringify(a));var x,C=(x=a,Object.keys(x[0]));console.log("tableColumns",C);var D=function(e){for(var t=[],a=0;a<e.length;a++){var n=Object(r.a)({},e[a]),l=e[a].Date,o=l.slice(6,10),s=l.slice(3,5),i=l.slice(0,2),c=new Date(o+"-"+s+"-"+i);n.Date=c,t.push(n)}return t}(a);console.log("dataWithDateFormat",D);var A=function(e){return b.a.sortBy(e,["Date","Datasource"])}(D);console.log("sorted data",A);var O=function(e,t){for(var a=[],n=0;n<t.length;n++){var l=t[n];b.a.isNumber(e[0][l])&&a.push(l)}return a}(A,C);console.log("metrics",O);var S=function(e,t){for(var a=[],n=0;n<t.length;n++){var l=t[n];b.a.isString(e[0][l])&&a.push(l)}return a}(A,C);console.log("dimension",S);var w=function(e,t){for(var a=[],n=0;n<t.length;n++){var l=t[n];b.a.isDate(e[0][l])&&a.push(l)}return a}(A,C);return console.log("dates",w),A[0].Date,A[1].Date,A[2].Date,A[3].Date,A[4].Date,console.log("selected values",h),l.a.createElement("div",{className:"App"},l.a.createElement(m.a,null,l.a.createElement(p,null),l.a.createElement(d,{dimensionsColumns:S,metricsColumns:O}),l.a.createElement(c.a,null,l.a.createElement(u.a,{sm:4},l.a.createElement("h2",null,"Selected values are: ",h),l.a.createElement(k,{styling:f,filterColumns:S,onChange:function(e,t){y(t.value),console.log("data.value",t.value)}})),l.a.createElement(u.a,{sm:8},l.a.createElement(E,{styling:f,coreData:A})))))};Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));s.a.render(l.a.createElement(D,null),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()}))}},[[154,1,2]]]);
//# sourceMappingURL=main.0c7851a1.chunk.js.map