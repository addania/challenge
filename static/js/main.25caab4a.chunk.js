(this.webpackJsonpchallenge=this.webpackJsonpchallenge||[]).push([[0],{10:function(e,t,n){e.exports=n(17)},15:function(e,t,n){},16:function(e,t,n){},17:function(e,t,n){"use strict";n.r(t);var a=n(0),l=n.n(a),r=n(8),o=n.n(r),c=(n(15),n(20)),i=(n(5),n(16),n(18)),m=n(19);function s(){return l.a.createElement("div",null,l.a.createElement(i.a,null,l.a.createElement(m.a,{sm:12},l.a.createElement("h2",{style:{margin:"30px 0px",textAlign:"left",color:"#7F6A93",borderBottom:"0.5px solid #E8E8EA",paddingBottom:"20px"}},"Adverity Coding Challenge"))))}function u(){var e={margin:"30px 0px",textAlign:"left",color:"#8DA1B9"};return l.a.createElement("div",null,l.a.createElement(i.a,null,l.a.createElement(m.a,{sm:4},l.a.createElement("h2",{style:e},"Filter")),l.a.createElement(m.a,{sm:8},l.a.createElement("h2",{style:e},"Graph"))))}var E=function(){return l.a.createElement("div",{className:"App"},l.a.createElement(c.a,null,l.a.createElement(s,null),l.a.createElement(u,null)))};Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));o.a.render(l.a.createElement(E,null),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()}))}},[[10,1,2]]]);
//# sourceMappingURL=main.25caab4a.chunk.js.map