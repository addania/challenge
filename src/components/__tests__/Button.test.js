import React from "react";
import ReactDOM from "react-dom";
import { Button } from "../Button.js";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button></Button>, div);
});

it("renders button correctly", () => {
  render(<Button></Button>);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Button></Button>);
  expect(getByTestId("button")).toHaveTextContent("Apply");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Button></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
