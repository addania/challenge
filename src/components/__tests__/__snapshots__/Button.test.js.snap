// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  data-testid="button"
  style={
    Object {
      "backgroundColor": "#8DA1B9",
      "borderRadius": "2px",
      "color": "white",
      "marginTop": "10px",
      "padding": "5px 0px",
      "position": "left",
      "width": "80px",
    }
  }
  type="submit"
>
  <span>
    Apply
  </span>
</button>
`;
